pkg/gpu/consumer.go

```diff
@@ -0,0 +1,166 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package gpu
+
+import (
+	"sync"
+	"time"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/gpu/model"
+	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	gpuebpf "github.com/DataDog/datadog-agent/pkg/gpu/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/process/monitor"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// CudaEventConsumer is responsible for consuming CUDA events from the eBPF probe, and delivering them
+// to the appropriate stream handler.
+type CudaEventConsumer struct {
```

Comment:
is that one specific for Cuda? are we going to have `NeutronEventConsumer` in the future?

Reply:
There's no guarantee things like the streams have to be managed in the same way, so it's possible we need a different consumer.